python/ray/serve/_private/request_router/request_router.py

```diff
@@ -526,16 +532,67 @@ def __init__(
                 "The number of request routing tasks in the router "
                 "that are undergoing backoff."
             ),
-            tag_keys=("app", "deployment", "actor_id"),
+            tag_keys=("app", "deployment", "actor_id", "application", "handle_source"),
         ).set_default_tags(
             {
+                # TODO(abrar): Remove "app" in future.
                 "app": self._deployment_id.app_name,
+                "application": self._deployment_id.app_name,
                 "deployment": self._deployment_id.name,
                 "actor_id": self_actor_id if self_actor_id else "",
+                "handle_source": self._handle_source.value,
             }
         )
         self.num_routing_tasks_in_backoff_gauge.set(self.num_routing_tasks_in_backoff)
 
+        # Queue wait time histogram: time request spent waiting in queue
+        # before being assigned to a replica.
+        self.queue_wait_time_ms_histogram = metrics.Histogram(
+            "serve_queue_wait_time_ms",
```

Comment:
i think the current name is slightly ambiguous
```suggestion
            "serve_request_router_fulfillment_time_ms",
```
or something similar